### PR TITLE
[CALCITE-3653] Support TableModify in ToLogicalConverter

### PR DESCRIPTION
**Pull request by xy2953396112**
Support TableModify convert to Logical in ToLogicalConverter

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/logical/ToLogicalConverter.java [hunk 1]
```diff
@@ -150,6 +151,15 @@ public ToLogicalConverter(RelBuilder relBuilder) {
       return LogicalCalc.create(visit(calc.getInput()), calc.getProgram());
     }
 
+    if (relNode instanceof TableModify) {
+      final TableModify tableModify = (TableModify) relNode;
+      final RelNode input = visit(tableModify.getInput());
+      return new LogicalTableModify(tableModify.getCluster(), tableModify.getTraitSet(),
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/rel/logical/ToLogicalConverter.java]: Shall we  use `LogicalTableModify#create` ?

Comment (xy2953396112) [reply core/src/main/java/org/apache/calcite/rel/logical/ToLogicalConverter.java]: Thanks for review, I think `new LogicalTableModify` and `LogicalTableModify#create` are equivalent. but it seems like that we use `LogicalTableModify#create` more better.

**Comment by jinxing64** [comment]
How about we update the title as 
`Support TableModify in ToLogicalConverter `

File: core/src/test/java/org/apache/calcite/rel/logical/ToLogicalConverterTest.java [hunk 2]
```diff
@@ -468,4 +468,19 @@ private void verify(RelNode rel, String expectedPhysical, String expectedLogical
...
+    String expectedPhysial = ""
+        + "JdbcToEnumerableConverter\n"
+        + "  JdbcTableModify(table=[[foodmart, employee]], operation=[INSERT], flattened=[true])\n"
+        + "    JdbcTableScan(table=[[foodmart, employee]])\n";
+    String expectedLogical = ""
+        + "LogicalTableModify(table=[[foodmart, employee]], "
+        + "operation=[INSERT], flattened=[true])\n"
+        + "  LogicalTableScan(table=[[foodmart, employee]])\n";
+    verify(rel, expectedPhysial, expectedLogical);
+  }
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/rel/logical/ToLogicalConverterTest.java]: Inline the `rel(sql)`.

Comment (xy2953396112) [reply core/src/test/java/org/apache/calcite/rel/logical/ToLogicalConverterTest.java]: > Inline the `rel(sql)`.

Thanks for you review, i have updated it.

**Comment by xy2953396112** [comment]
@jinxing64 @danny0405 @chunweilei  Thanks for you review.

**Pull request merged**